models/articles.models.js

```diff
@@ -10,4 +10,25 @@ function fetchArticleById(id) {
         })
 }
 
-module.exports = { fetchArticleById }
\ No newline at end of file
+function fetchAllArticles() {
+    return db.query(`
+        SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, 
+        COUNT(comments.comment_id) AS comment_count 
+        FROM articles
+        LEFT JOIN comments 
+        ON articles.article_id = comments.article_id
+        GROUP BY articles.article_id
+        ORDER BY articles.created_at DESC`)
+        .then(({rows}) => {
+            if (rows.length === 0) {
+                return Promise.reject({status: 404, msg: 'articles not found'})
+            }
+            rows = rows.map((article) => ({
```

Comment:
We shouldn't need to do this. The comment count should be on each article object 

Each article should now have the following:
author 
title
article_id
topic
article_img_url
created_at
votes
comment_count